### PR TITLE
mailbox: unify sram fw regs

### PR DESCRIPTION
**Pull request by abonislawski**
This patch will move all sram fw regs offsets (only offsets) to a commpon
mailbox.h

### CODE REVIEW

File: src/include/sof/lib/mailbox.h [hunk 1]
```diff
@@ -16,6 +16,16 @@
...
 
+/* SRAM window 0 FW "registers" */
+#define SRAM_REG_ROM_STATUS			0x0
+#define SRAM_REG_FW_STATUS			0x4
+#define SRAM_REG_FW_TRACEP			0x8
+#define SRAM_REG_FW_IPC_RECEIVED_COUNT		0xc
+#define SRAM_REG_FW_IPC_PROCESSED_COUNT		0x10
+#define SRAM_REG_FW_TRACEP_SLAVE_CORE_BASE	0x14
+#define SRAM_REG_FW_END \
+	(SRAM_REG_FW_TRACEP_SLAVE_CORE_BASE + (PLATFORM_CORE_COUNT - 1) * 0x4)
```
Comment (mmaka1) [inline src/include/sof/lib/mailbox.h:27]: Might be better to define 4 (at least, 8 might be better long term?) slave core slots statically to make sure that subsequent regs have stable, platform independent offset.

**Comment by mmaka1** [comment]
Is this header detected by the checkpatch as part of ABI definition?

**Comment by lgirdwood** [comment]
As @mmaka1 has pointed out these are ABI that the kernel or userspace tools  may use to diagnose any issues or status.

File: src/include/sof/lib/mailbox.h (hunk 1 above)
Comment (abonislawski) [reply src/include/sof/lib/mailbox.h]: ok, changed to 8

**Comment by abonislawski** [comment]
@mmaka1 @lgirdwood in checkpatch abi check is defined for:
src/include/ipc/
src/include/kernel/
src/include/user/

I can add src/include/sof/lib/mailbox.h but maybe it will be better to create a new file include/kernel/mailbox.h?
please check newest changes for this

**Comment by abonislawski** [comment]
> This is a better location, indicates the purpose and consumer. Could you dox the defs as well? (just a brief documentation that may be developed further)

@mmaka1  added general \brief description

**Comment by lgirdwood** [comment]
@abonislawski can you check CI

**Comment by abonislawski** [comment]
@lgirdwood pr-device-test not related to this PR, pr-checkpatch ABI warnings

**Comment by marc-hb** [comment]
> pr-checkpatch ABI warnings

Not just ABI warnings but admittedly hard to spot before #3158

**Pull request merged**